### PR TITLE
Fixes "IndexError: list index out of range" in add_partonomy.py.

### PR DESCRIPTION
**Pull request by lchoy**
Fixes "IndexError: list index out of range" in add_partonomy.py when the `origin_samples` field is not available.
Fixes CAT-529.

### CODE REVIEW

**Comment by yuanzhou** [comment]
@lchoy these are the corresponding PROD reindex from 3/11, affecting 4 samples and 5 datasets. Which is the one you mentioned with `Invalid` status that we'll need to fix at the data level?
```
[2024-03-11 03:17:47] ERROR in hubmap_translator:671: list index out of range
[2024-03-11 03:17:47] ERROR in hubmap_translator:674: Exception encountered during executing call_indexer() for uuid: d746ed2034f6d5fd098d2af657634b51, entity_type: Sample
[2024-03-11 03:17:48] ERROR in hubmap_translator:671: list index out of range
[2024-03-11 03:17:48] ERROR in hubmap_translator:674: Exception encountered during executing call_indexer() for uuid: b853016f0e0c76895cb68151683f5dad, entity_type: Sample
[2024-03-11 03:17:48] ERROR in hubmap_translator:671: list index out of range
[2024-03-11 03:17:48] ERROR in hubmap_translator:674: Exception encountered during executing call_indexer() for uuid: 057e978bfd24e33e86227e357d40b35c, entity_type: Sample
[2024-03-11 03:17:48] ERROR in hubmap_translator:671: list index out of range
[2024-03-11 03:17:48] ERROR in hubmap_translator:674: Exception encountered during executing call_indexer() for uuid: 9dfda76c22fbfd238c62b24686f29844, entity_type: Sample
[2024-03-11 03:21:22] ERROR in hubmap_translator:671: list index out of range
[2024-03-11 03:21:22] ERROR in hubmap_translator:674: Exception encountered during executing call_indexer() for uuid: fe2babe40b7f17034776ff3031ca0cb5, entity_type: Dataset
[2024-03-11 03:21:22] ERROR in hubmap_translator:671: list index out of range
[2024-03-11 03:21:22] ERROR in hubmap_translator:674: Exception encountered during executing call_indexer() for uuid: e8d3bbcba4977dbcfc480c5dc6526e50, entity_type: Dataset
[2024-03-11 03:21:22] ERROR in hubmap_translator:671: list index out of range
[2024-03-11 03:21:22] ERROR in hubmap_translator:674: Exception encountered during executing call_indexer() for uuid: 7e16ff08f6497b533a9a349e732e95cd, entity_type: Dataset
[2024-03-11 03:21:22] ERROR in hubmap_translator:671: list index out of range
[2024-03-11 03:21:22] ERROR in hubmap_translator:674: Exception encountered during executing call_indexer() for uuid: f2d4cf63e8080d4d2c2821c85d3ff9d6, entity_type: Dataset
[2024-03-11 03:37:39] ERROR in hubmap_translator:671: list index out of range
[2024-03-11 03:37:39] ERROR in hubmap_translator:674: Exception encountered during executing call_indexer() for uuid: 74913862edf83f72cdb6402a861f744e, entity_type: Dataset
```

File: src/hubmap_translation/addl_index_transformations/portal/add_partonomy.py [hunk 1]
```diff
@@ -17,8 +17,10 @@
 
 
 def _get_organ_iri(doc):
-    two_letter_code = doc.get('origin_samples', [{}])[0].get('organ')
-    return _two_letter_to_iri.get(two_letter_code)
+    origin_samples = doc.get('origin_samples', [])
+    first_sample = origin_samples[0] if origin_samples else {}
+    two_letter_code = first_sample.get('organ', None)
+    return _two_letter_to_iri.get(two_letter_code) if two_letter_code else None
```
Comment (john-conroy) [inline src/hubmap_translation/addl_index_transformations/portal/add_partonomy.py:23]: Can we add a call to `_log_transformation_error`?

**Review by john-conroy — COMMENTED** [review]
Last I heard we were waiting for Brendan to check with the data providers as to whether these samples/datasets could be deleted.

I'm not sure if the detail pages would break if a dataset or sample does not have a sample with an organ in `origin_samples` or values in `origin_samples_unique_mapped_organs`. We haven't fully typed those components.

What is the motivation here?

File: src/hubmap_translation/addl_index_transformations/portal/add_partonomy.py (hunk 1 above)
Comment (lchoy) [reply src/hubmap_translation/addl_index_transformations/portal/add_partonomy.py]: done!

**Review by john-conroy — APPROVED** [review]

**Review by yuanzhou — APPROVED** [review]

**Pull request merged**